lesson13/js/main.js

```diff
@@ -0,0 +1,154 @@
+	createCloseBtn.textContent = '×';
+
+	createMenuContainer.setAttribute('id', 'js-menu_container');
+
+	modal.appendChild(createMenuBtn);
+	modal.appendChild(createCloseBtn);
+	modal.appendChild(createMenuContainer);
+	modalWindow.appendChild(modal);
+	document.body.appendChild(modalWindow);
+
+	loadMenu();
+	closeModalWindow();
+}
+
+//open Modal
+const openModalBtn = document.getElementById('js-openModalBtn');
+
+openModalBtn.addEventListener(
+	'click',
+	() => {
+		const modalBg = document.createElement('div');
+		modalBg.id = 'js-bgFixed';
+		modalBg.classList.add('el_modalBg');
+		document.body.appendChild(modalBg);
+		createModalWindow();
+		document.body.style.position = 'fixed';
+	},
+	false
+);
+
+function loadMenu() {
+	const loadMenuBtn = document.getElementById('js-loadmenuBtn');
+	loadMenuBtn.addEventListener(
+		'click',
+		(event) => {
+			callMenuContents();
+			event.stopPropagation();
+		},
+		false
+	);
+}
+
+function closeModalWindow() {
+	const closeModlBtn = document.getElementById('js-closeModal');
+
+	closeModlBtn.addEventListener(
+		'click',
+		() => {
+			const modalContainer = document.getElementById('js-modalContainer');
+			const modalWindow = document.createElement('div');
+			const bgFixed = document.getElementById('js-bgFixed');
+			removeInnerHTML(modalWindow);
+			removeInnerHTML(menuLists);
+			document.body.removeAttribute('style');
+			bgFixed.remove();
+			modalContainer.remove();
+		},
+		false
+	);
+}
+
+//CREATE MENU contents
+function showLoadingImg() {
+	const menuContainer = document.getElementById('js-menu_container');
+	const createLoadingBox = document.createElement('div');
+	createLoadingBox.classList.add('ly_loading');
+	createLoadingBox.id = 'js-loading';
+	const loadingImg = document.createElement('img');
+	loadingImg.src = 'images/loading-circle.gif';
+	loadingImg.classList.add('el_loadingImg');
+	menuContainer.appendChild(createLoadingBox).appendChild(loadingImg);
+}
+
+function hideLoadingImg() {
+	const loading = document.getElementById('js-loading');
+	loading.remove();
+}
+
+function removeInnerHTML(element) {
+	element.innerHTML = '';
+}
+
+//menulist
+function createMenuList(data) {
+	const menuContainer = document.getElementById('js-menu_container');
+	const fragment = document.createDocumentFragment();
+	removeInnerHTML(menuLists);
+
+	data.forEach((value) => {
+		const listItem = document.createElement('li');
+		const listImg = document.createElement('img');
+		listImg.src = value.img;
+		listImg.alt = value.alt;
+		listImg.width = 40;
+		listImg.height = 40;
+
+		const listAnchor = document.createElement('a');
+		listAnchor.href = value.a;
+		listAnchor.textContent = value.text;
+		fragment.appendChild(listItem).appendChild(listAnchor).appendChild(listImg);
+	});
+	menuContainer.appendChild(menuLists).appendChild(fragment);
+}
+
+const MENU_DATA_URL = 'https://jsondata.okiba.me/v1/json/Aticp210816214718';
+async function fetchMenuContents() {
+	showLoadingImg();
```

Comment:
【IMO】
I think that the job of this function is only to fetch the fetch data, and the execution location of showLoadingImg () may be other location.

Reply:
Thank you for review!
It’s just as you said.
Changed to the callMenuContents () function that calls the menu.
Thanks to that, it became very natural.

[03fe91d](https://github.com/nyattsu72/hanson-frontend-cource/pull/18/commits/03fe91d19c84aa348231034b40697312d65525e7)